### ISSUE TITLE
Build fails when using #" char sequence within the string value.

### ISSUE THREAD

**Issue by JaviSFH** [SwiftGen/SwiftGen#986]
**Environment**: 

- SwiftGen 6.6.1
- Xcode 13.4.1
- Targeting iOS 15+

We use HTML tags in our string values so that the app can parse them and convert them to actual text styles. From the SwiftGen version 6.6.0 on, we're facing an issue with the new `fallback` generated parameter in the generated string file when using the HTML tag `font color="#12345"`. For instance, for the string `"text_with_color" = "<font color="#333333">Styled text</font>";` the generated string looks like this:

`internal static let failed = Strings.tr("Localization", "text_with_color", fallback: #"<font color="#333333">Styled text</font>"#)`

**Problem:** 

`fallback` parameter uses Swift's raw strings with the delimiter `#` which is colliding with the `"#` from the string, causing the project to fail at compile time.

**Possible solution**

Add more hash symbols as delimiter. The generated string below would compile perfectly. 

`internal static let failed = Strings.tr("Localization", "text_with_color", fallback: ##"<font color="#333333">Styled text</font>"##)`

Thank you.

**Comment by djbe** [comment]
Ah that's unfortunate, that combination of `"` and `#` 🤦  As a temporary workaround, you can use `'` in your HTML (should work), or use a custom template for now.

**Comment by teejayhh** [comment]
Also if the fallback is used then things like `\n` stop working. The fallback should work the exact same way the actual value works. 

I fixed my issues by doing this in the structured stencil 

```
  {% set translation %}{{string.translation|replace:'"', '\"'}}{% endset %} <-------
  {% if string.types %}
  {{accessModifier}} static func {{string.name|swiftIdentifier:"pretty"|lowerFirstWord|escapeReservedKeywords}}({% call parametersBlock string.types %}) -> String {
    return {{enumName}}.tr("{{table}}", "{{string.key}}", {%+ call argumentsBlock string.types %}, fallback: "{{translation}}")
  }
  {% elif param.lookupFunction %}
  {{accessModifier}} static var {{string.name|swiftIdentifier:"pretty"|lowerFirstWord|escapeReservedKeywords}}: String { return {{enumName}}.tr("{{table}}", "{{string.key}}", fallback: "{{translation}}") }
  {% else %}
  {{accessModifier}} static let {{string.name|swiftIdentifier:"pretty"|lowerFirstWord|escapeReservedKeywords}} = {{enumName}}.tr("{{table}}", "{{string.key}}", fallback: "{{translation}}")
  {% endif %}
```

**Comment by djbe** [comment]
Not sure about that? We have tests with multiline keys, values & comments:
- Strings file: https://github.com/SwiftGen/SwiftGen/blob/stable/Sources/TestUtils/Fixtures/Resources/Strings/LocMultiline.strings
- Generated code: https://github.com/SwiftGen/SwiftGen/blob/develop/Sources/TestUtils/Fixtures/Generated/Strings/structured-swift5/multiple.swift#L107-L138

**Comment by teejayhh** [comment]
Perhaps you could try this in your test 

en
```
"somekey" = "hello \nWorld" -> translation should break the line
"hello" = "\"Quotes\""
"hello2" = "Dudes"
```

en-au
```
"hello2" = "mates"
```

if you now have the AU region activated it will pull the Fallback from - en for the key `somekey` and `hello`. which won't work. I have like 300 translations for en but only 20 for en_au so the fallback works great for me but fails for those cases above. (I don't like duplication of the language files for regional changes which I have 3 of (US(CA)/AU/RestOfTheWorld))

I changed the  `#"{{string.translation}}"#` -> just `"{{string.translation}}"` which made it necessary to do this 
```
{{string.translation|replace:'"', '\"'}}
```
because otherwise `\"` escaped double quotes would break the generated source. Look, it's a very specific issue perhaps not many people have. I can make it work with a custom stencil but the default one is broken for my translations. 

`Extended String Delimiters` work this way but they they do not behave the same way as if you would pull the normal translation vs fallback (one is escaped and one is not).

**Comment by narlei** [comment]
I'm having a problem too with breaking lines with escape `\n`. 
```
"key" = "My \n value"
```

This key shows like:
```
My
value
```

But fallback is using the raw marker `#` and evaluating the `\n` then shows:
```My\nvalue```

I think that the best way to solve it by now is creating a config to disable the fallback.

**Comment by djbe** [comment]
Aaaah I think I get it now, something I completely missed is that in Swift *raw* strings, escape sequences need to have the delimiter, so for `###"…"###`, the escape sequence for `\n` becomes `\###n`. 

Sure our generated code is valid and compiles now, but the runtime value isn't what it should be… ☹️

**Comment by djbe** [comment]
Fixed in 6.6.2

**Comment by gerchicov-vg** [comment]
@djbe 
https://github.com/SwiftGen/SwiftGen/issues/1150
seems you have fixed these problems for values only but what if keys contain such special symbols?
